CONTRIBUTING_A_NEW_QUEST.md

```diff
@@ -0,0 +1,390 @@
+
+It is a message displayed to user, code here passes a [reference](https://developer.android.com/guide/topics/resources/string-resource) to the string. You can change it to the new, not yet existing one and use a built in tool to place text.
+
+Actual strings sit in [app/src/main/res/values/strings.xml](app/src/main/res/values/strings.xml)
+
+There are additional files with text, but the text is translated with a completely separately worflow and your changes will modify solely this one file, leaving other text files untouched.
+
+#### Form
+
+`override fun createForm() = YesNoQuestAnswerFragment()`
+
+Form defines interface used by mappers.
+
+In this case, the simplest possible is used.
+
+But sometimes more complex ones are needed, see for example [AddBridgeStructure.kt](https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/bridge_structure/AddBridgeStructure.kt) 
+
+`override fun createForm() = AddBridgeStructureForm()`
+
+
+With form defined in [AddBridgeStructureForm](app/src/main/java/de/westnordost/streetcomplete/quests/bridge_structure/AddBridgeStructureForm.kt)
+### applyAnswerTo
+```
+    override fun applyAnswerTo(answer: Boolean, changes: StringMapChangesBuilder) {
+        changes.add("indoor", answer.toYesNo())
+    }
+``` 
+
+This code is responsible for modifying `StringMapChangesBuilder` object (passed by reference).
+
+Actions may include (examples from various quests):
+
+- `changes.add("indoor", answer.toYesNo())` - adds a new tag, it must not be present already
+- `changes.addOrModify("material", newMaterial)` - add a new tag if missing, override existing value if present
+- `changes.modify("sport", values)` - modify existing tag, value must be present
+- `changes.delete("amenity")` - remove key that must be present
+- `changes.deleteIfExists("stile")` - remove key if present
+- `changes.updateWithCheckDate("lit", answer.toYesNo())` - acts as `addOrModify` but will also update survey date and can be used in cases where tag value is not changing
+
+See other ones defined in [StringMapChangesBuilder.kt](de/westnordost/streetcomplete/data/osm/edits/update_tags/StringMapChangesBuilder.kt)
+
+### Extras
+
+Info listed above must be supplied by every quest. But there are also several optional fields. This specific qest has
+```
+    override fun getHighlightedElements(element: Element, getMapData: () -> MapDataWithGeometry) =
+        getMapData().filter("nodes with emergency = defibrillator")
+```
+
+which causes nearby `emergency = defibrillator` nodes to be shown (icons used are defined in [PinIcons.kt](app/src/main/java/de/westnordost/streetcomplete/map/PinIcons.kt))
+
+See [also other optional properties](de/westnordost/streetcomplete/data/osm/osmquests/OsmElementQuestType.kt).
+
+### Adding quest icon
+
+It is OK to submit a quest without own icon, using any icon as a placeholder.
+
+But it would be even better to include also icon.
```

Comment:
I am not convinced that it is an improvement, it seems to be using more complex sentence structure without a benefit.

But I am not a native speaker...

Reply:
You're not supposed to start a sentence with a but (grammatically).

Also this doesn't "scan" right:
"it would be even better to include also icon."

It should be this.
"it would be even better to also include an icon."

You could also drop the but and put "However..."

Reply:
@peternewman Hmmm. I went looking though this and found numerous claims that it is actually OK to start from "but".

Not entirely convinced here, but I am not a native speaker so I do really trust myself either.